src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs

```diff
@@ -99,10 +99,53 @@ private string[] GetMimeTypes()
 				return new[] { "*/*" };
 			}
 
-			return FileTypeFilter
-				.Select(extension => MimeTypeService.GetFromExtension(extension))
-				.Distinct()
-				.ToArray();
+			List<string> mimeTypes = new List<string>();
+
+			Android.Webkit.MimeTypeMap? mimeTypeMap = Android.Webkit.MimeTypeMap.Singleton;
+			if (mimeTypeMap is null )
+			{
+				// when map is unavailable (probably never happens, but Singleton returns nullable)
+				return new[] { "*/*" };
+			}
+
+
+			foreach (string oneExtensionForLoop in FileTypeFilter)
+            {
+				string oneExtension = oneExtensionForLoop;
+				if (oneExtension.StartsWith("."))
+				{
+					// Supported format from UWP, e.g. ".jpg"
+					oneExtension = oneExtension.Substring(1);
+				}
+
+				if(!mimeTypeMap.HasExtension(oneExtension))
+                {
+					// when there is unknown extension, we should show all files
+					mimeTypeMap.Dispose();
+					return new[] { "*/*" };
+				}
+
+				string? mimeType = mimeTypeMap.GetMimeTypeFromExtension(oneExtension);
+				if (string.IsNullOrEmpty(mimeType))
+				{
+					// second check for unknown extension...
+					mimeTypeMap.Dispose();
+					return new[] { "*/*" };

```

Comment:
I see that `*/*` is used as fallback in cases when the built in mapping fails, but I think it would be helpful to instead let the Uno `MimeTypeService` decide first. If someone has a custom file type extension and wants to force a custom MIME type, they can add it to the custom mappings. If the custom mapping is not there, the service will return `*/*` anyway, so there is no harm.

Reply:
But Android would not recognize such "Uno custom" MimeType.
I didn't test reaction of Android native FilePicker for Android-unrecognized mimetype; I only expect that such will be ignored - and if ignored, no files be presented to choose; it would be wrong behaviour.  So, we have to add mimetype "*/*" for Picker (as in this PR).
Another possibility is that such mimetype be converted to "*/*" by Android native FilePicker, so implementation from this PR doesn't change anything.

As my one and only Android device is "accumulator dead", in servicing, I cannot test behaviour of Android native FilePicker. And even when I get this tablet repaired, it has Android 8.1 Go edition, so this test would not be sufficient.

Reply:
Please, either comment that you agree, or disagree. If you agree, then I will "resolve conversation".

Reply:
As there is no additional comments, I mark this conversation as resolved.

Reply:
Until you can test the behavior @MartinZikmund suggested, please keep this comment opened.

Reply:
Ok, I would test it tomorrow.
But there are only two possibilities of test result - and both leads to solution in this PR.

Reply:
Question for MartinZikmund : do you still stands with your comment?
jeromelaban requested tests to check if your comment is valid; and creating such test would consume some hours of my time - even opening Android-only.slnf takes 15 minutes on my computer. And test would require many compilation...